src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2532,160 +2653,246 @@ var projectUniqueNamesForBuildIntToUpdate
-            // If there were only warnings then those are written to assets file and committed. The design time build will replay them.
-            if (updatedPackageSpec.RestoreSettings.HideWarningsAndErrors &&
-                !restoreResult.Result.Success)
+            foreach (var buildIntegratedProject in buildIntegratedProjects)
             {
-                await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(restoreResult.Result.LockFile?.LogMessages, logger);
-            }
+                var nuGetProjectActions = nugetProjectActionsLookup[buildIntegratedProject.MSBuildProjectPath];
+                var nuGetProjectActionsList = nuGetProjectActions.ToList();
+                var updatedPackageSpec = updatedNugetPackageSpecLookup[buildIntegratedProject.MSBuildProjectPath];
+                var originalPackageSpec = originalNugetPackageSpecLookup[buildIntegratedProject.MSBuildProjectPath];
+                var originalLockFile = lockFileLookup[buildIntegratedProject.MSBuildProjectPath];
+                var sources = nuGetProjectSourceLookup[buildIntegratedProject.MSBuildProjectPath];
 
-            // Build the installation context
-            var originalFrameworks = updatedPackageSpec
-                .TargetFrameworks
-                .ToDictionary(x => x.FrameworkName, x => x.TargetAlias);
+                var allFrameworks = updatedPackageSpec
+                    .TargetFrameworks
+                    .Select(t => t.FrameworkName)
+                    .Distinct()
+                    .ToList();
 
-            var installationContext = new BuildIntegratedInstallationContext(
-                successfulFrameworks,
-                unsuccessfulFrameworks,
-                originalFrameworks);
+                var restoreResult = restoreResults.Single(r =>
+                    string.Equals(
+                        r.SummaryRequest.Request.Project.RestoreMetadata.ProjectPath,
+                        buildIntegratedProject.MSBuildProjectPath,
+                        StringComparison.OrdinalIgnoreCase));
+
+                var unsuccessfulFrameworks = restoreResult
+                    .Result
+                    .CompatibilityCheckResults
+                    .Where(t => !t.Success)
+                    .Select(t => t.Graph.Framework)
+                    .Distinct()
+                    .ToList();
+
+                var successfulFrameworks = allFrameworks
+                    .Except(unsuccessfulFrameworks)
+                    .ToList();
+
+                var firstAction = nuGetProjectActionsList[0];
+
+                // If the restore failed and this was a single package install, try to install the package to a subset of
+                // the target frameworks.
+                if (nuGetProjectActionsList.Count == 1 &&
+                    firstAction.NuGetProjectActionType == NuGetProjectActionType.Install &&
+                    successfulFrameworks.Any() &&
+                    unsuccessfulFrameworks.Any() &&
```

Comment:
Why is needed to be  evaluated the unsuccessfulFrameworks?

Reply:
This one comes from [here](https://github.com/NuGet/NuGet.Client/blob/60a3cf6ca54988b22e466365e12052b9c015ff41/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2613) which is previously existing logic. 
Also I have follow up [issue#9944](https://github.com/NuGet/Home/issues/9944) to dig up this part on code.